pkg/api/validation/validation.go

```diff
@@ -2562,6 +2572,17 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(specPath.Child("activeDeadlineSeconds"), newPod.Spec.ActiveDeadlineSeconds, "must not update from a positive integer to nil value"))
 	}
 
+	// validate updated spec.PriorityClassName.
+	if newPod.Spec.PriorityClassName != oldPod.Spec.PriorityClassName && len(newPod.Spec.PriorityClassName) > 0 {
```

Comment:
You are disallowing "foo" -> "bar", but "foo" -> "" is OK?  why?  If this is right, comment why, but explain it first?